docs/source/hyper-parameter-search.rst

```diff
@@ -15,7 +15,7 @@ The underlying estimator will need to be able to train on each cross-validation
 See :ref:`hyperparameter.drop-in` for more.
 
 If your data is large and the underlying estimator implements ``partial_fit``, you can
-Dask-ML's :ref:`*incremental* hyperparameter optimizers <hyperparameter.incremental>`.
```

Comment:
I'm starting to doubt the "two kinds of hyperparameter optimization estimators" I laid out on line 6.

After our conversation the other day, it sounds like there are (at least) 3.

1. Static search: All combinations to be tried are specified up front. The full dataset (or a CV split of the full dataset) is used on each training call.
2. Adaptive search: Regardless of exactly which data is used (CV split of full dataset, or a batch), the salient feature of adaptive search is that models are prioritized according to past performance. 
3. Incremental search: Regardless of how models are chosen for training (static or adaptive), the salient feature of Incremental search is that models are trained on *batches* of data passed to `_parital_fit`. The underlying estimator *must* implement `partial_fit`.

Our current implementation is a mix of 2 and 3. Does that framing accord with other's thoughts? Can we think of a clear way to explain that? The docs below do a decent job I think. This line, though, I think emphasizes adaptive too much given the framing above (you have a large dataset).

Reply:
Maybe a clearer way of explaining it:

> There are two dimensions to searches:
>
> 1. Does the search use `partial_fit` or `fit`?
> 2. Is the search passive or adaptive? That is, does it use previous evaluations to select which models to train further?

Your categories fall pretty nicely into this (maybe renaming "static" to "passive").